solution/ui/regulations/eregs-vite/src/components/resources/ResourceExportBtn.vue

```diff
@@ -0,0 +1,121 @@
+                    fill="white" />
+            </svg>
+        </button>
+    </div>
+
+</template>
+
+<script>
+
+import exportFromJSON from 'export-from-json'
+import {
+    getSupplementalContentV3,
+
+} from "../../utilities/api";
+export default {
+    name: "ResourceExportBtn",
+    props: {
+        searchQuery: {
+            type: String,
+            default: ""
+        },
+        partDict: {
+            type: Object,
+            default: () => {}
+        },
+        categories: {
+            type: Array,
+            default:() => []
+        },
+        supCount: {
+            type: Number,
+            default: 0
+        }
+    },
+    data: {
+        supplementalContent: []
+    },
+    computed: {
+
+    },
+    methods: {
+        async createCSV() {
+            this.supplementalContent = await this.getSupplementalContent()
+            const supSort = this.supplementalContent.map(cont => {
+                const content = {}
+                content.category= cont.category.name
+                content.date = cont.date
+                content.description = cont.description
+                content.document_number = cont.document_number
+                content.name = cont.name
+                content.url = cont.url
+                content.Related_Regulations = this.formatLocations(cont.locations)
+                return content
+            }
+            )
+            const fileName = 'Resources'
+            const exportType = exportFromJSON.types.csv;
+            exportFromJSON({ data: supSort, fileName, exportType })
+        },
+
+        formatLocations(locations) {
+            let locString = "";
+            for (const location in locations) {
+                if (locations[location].type == "section") {
+                    locString = `${locString} CFR ${locations[location].title} ${locations[location].part}.${locations[location].section_id},`
+                }
+                else {
+                    locString = `${locString} ${locations[location].title} Subpart ${locations[location].subpart_id},`
+                }
+            }
+            return locString
+        },
+        async getSupplementalContent() {
+            let supNum = 0;
+            let page = 1;
+            let content = []
+            let responseContent = []
+            while (supNum < this.supCount) {
+                content = getSupplementalContentV3({
+                    page,
+                    partDict: this.partDict,
+                    categories: this.categories,
+                    q: this.searchQuery
+                });
+
+                page += 1
+                supNum += 100;
+                responseContent = responseContent.concat(content.results)
+
+            }
+            return responseContent
+        },
+    },
+}
+</script>
+<style scoped>
+.csv-button {
```

Comment:
Is this `csv-button` class being used anywhere?

Reply:
removed